### PR TITLE
TelemetryProcessor - invoke initialize if ITelemetryModule interface also implemented by telemetryprocessor

### PR DESCRIPTION
**Pull request by cijothomas**

### CODE REVIEW

**Comment by cijothomas** [comment]
@upendras

**Comment by SergeyKanzhelev** [comment]
Can you please resolve conflicts.
Looks good otherwise.

File: src/Core/Managed/Shared/Extensibility/Implementation/TelemetryProcessorChain.cs [hunk 1]
```diff
@@ -20,11 +20,33 @@ public sealed class TelemetryProcessorChain
...
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TelemetryProcessorChain" /> class by using the given list elements.
+        /// Marked internal, as clients should use TelemetryProcessorChainBuilder to build the processing chain.
+        /// </summary>
+        internal TelemetryProcessorChain(IEnumerable<ITelemetryProcessor> telemetryProcessors)
+        {
+            foreach (var item in telemetryProcessors)
+            {
+                this.telemetryProcessors.Add(item);
```
Comment (upendras) [inline src/Core/Managed/Shared/Extensibility/Implementation/TelemetryProcessorChain.cs]: nit - use AddRange() instead?

File: src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs [hunk 2]
```diff
@@ -252,7 +252,7 @@ private static void InitializeComponents(TelemetryConfiguration configuration)
             InitializeComponent(configuration.TelemetryChannel, configuration);            
             InitializeComponents(configuration.TelemetryInitializers, configuration);
             InitializeComponents(configuration.ContextInitializers, configuration);
-
+            InitializeComponents(configuration.TelemetryProcessors.TelemetryProcessors, configuration);
```
Comment (upendras) [inline src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs]: I guess TelemetryProcessorChain should implement ITelemetryModule and in the initialize method, initialize the ones that implement ITelemetryModule

**Pull request merged**